File: go/24_tree/TreeNode.go
package _4_tree

import "fmt"

type Node struct {
	data  interface{}
	left  *Node
	right *Node
}

func NewNode(data interface{}) *Node {
	return &Node{data: data}
}

func (this *Node) String() string {
	return fmt.Sprintf("v:%+v, left:%+v, right:%+v", this.data, this.left, this.right)
}

File: php/24_tree/TreeNode.php
<?php

namespace Algo_24;

class TreeNode
{

    /**
     * 节点中的数据
     * @var [type]
     */
    public $data;

    /**
     * 左节点
     * @var [type]
     */
    public $left;

    /**
     * 右节点
     * @var [type]
     */
    public $right;

    /**
     * [__construct description]
     * @param [type] $data [description]
     */
    public function __construct($data = null)
    {
        $this->data = $data;
        $this->left = null;
        $this->right = null;
    }
}